### ISSUE TITLE
Validation messages missing from datepicker form elements

### ISSUE THREAD

**Issue by samtay** [magento/magento2#3795]
## Steps to reproduce
1. Install Magento from `develop` branch.
2. Edit a CMS Page 
3. Choose _Custom Design From_ and _Custom Design To_ that are faulty (poorly formatted or such that the end date occurs **before** the start date.
## Expected result
1. See error message generated from `validation.js`
## Actual result
1. In some situations (bad start/end logic or formatting), the validation is caught in JS but you can't see the message on the page. So you can keep clicking "Save" and it will keep focusing on the problematic field, but there is no error message there. It does not exist anywhere in the DOM.
   ![2016-03-15-141526_1765x854_scrot](https://cloud.githubusercontent.com/assets/7246591/13788968/6b4045e8-eab8-11e5-9b88-89360c87991b.png)
2. For other situations (not sure how to replicate), validation is not caught in JS and instead is caught in php POST.
   ![2016-03-15-141028_1755x641_scrot](https://cloud.githubusercontent.com/assets/7246591/13788816/b0e49afa-eab7-11e5-99ef-2829110e42d1.png)

**Comment by shiftedreality** [comment]
Hi @SamTay 

Looks like you cloned 2.0 branch but not develop. Latest CMS Page has different layout from provided. Could you please verify this issue still actual for you?

**Comment by samtay** [comment]
You are correct - my screenshots are from EE. However, bad news is that I just cloned develop from this repo and there is a larger problem. The validation is missing, and you can now save a `custom_design_from` date **after** a `custom_design_to` date.

I just looked into this to see if I could send you a quick pull request, as you are missing some of the validation classes. In fact, the recent changes removed date-range validation across the entire framework (for example, [cms page design dates](https://github.com/magento/magento2/blob/aad945ce88bc128f209d39e2e09e3553d8c04e7c/app/code/Magento/Cms/view/adminhtml/ui_component/cms_page_form.xml#L247)).

I thought this would be easy to fix by leveraging existing `validation.js` rules like this:

``` xml
<item name="validation" xsi:type="array">
    <item name="validate-date" xsi:type="boolean">true</item>
    <item name="validate-date-range" xsi:type="boolean">true</item>
</item>
```

and was trying to figure out how to add the class  `date-range-custom_theme-from` to the form element... I can't figure out how to add classes to form fields through the UI components.

BUT this may be a pointless exercise because it looks like the validate-date-range has been removed from the javascript altogether: 
![2016-03-17-142837_1907x621_scrot](https://cloud.githubusercontent.com/assets/7246591/13856688/bfa0f2b4-ec4c-11e5-9d1b-835cde12c50a.png). This is confusing because I still see the rule in [`validation.js`](https://github.com/magento/magento2/blob/aad945ce88bc128f209d39e2e09e3553d8c04e7c/lib/web/mage/validation.js#L849-L870) but it doesn't appear to be available on the frontend.

**Comment by piotrekkaminski** [comment]
@elenleonova can you look into it? any reason for validation removal?

**Comment by magento-engcom-team** [comment]
@samtay, thank you for your report.
We've created internal ticket(s) MAGETWO-82232 to track progress on the issue.

**Comment by sadeeshmca** [comment]
I am working on this at #dmcdindia

**Comment by magento-engcom-team** [comment]
@sadeeshmca thank you for joining. Please accept team invitation [here](https://github.com/orgs/magento/invitation) and self-assign the issue.

**Comment by thanura-bi** [comment]
I'm working on #cd-balance

**Comment by magento-engcom-team** [comment]
@thanura-bi thank you for joining. Please accept team invitation [here](https://github.com/orgs/magento/invitation) and self-assign the issue.

**Comment by magento-engcom-team** [comment]
Hi @ravi-chandra3197. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if you want to validate it one more time, please, go though the following instruction:
- [x] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [x] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by ravi-chandra3197** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @ravi-chandra3197. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @ravi-chandra3197, here is your [Magento instance](https://i-3795-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-3795-2-3-develop.instances.magento-community.engineering/admin](https://i-3795-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by magento-engcom-team** [comment]
Hi @samtay. Thank you for your report. 
The issue has been fixed in magento/magento2#21397 by @ravi-chandra3197 in 2.3-develop branch
Related commit(s):
 - [8bb401e0ff8641d25e4fd70e935cff5c67f2def7](https://github.com/magento/magento2/commit/8bb401e0ff8641d25e4fd70e935cff5c67f2def7)

The fix will be available with the upcoming 2.3.3 release.

**Comment by magento-engcom-team** [comment]
Hi @samtay. Thank you for your report. 
The issue has been fixed in magento/magento2#23002 by @ravi-chandra3197 in 2.2-develop branch
Related commit(s):
 - [fe374b929cf074a6d1ec3fbe3b48f70d615e4f82](https://github.com/magento/magento2/commit/fe374b929cf074a6d1ec3fbe3b48f70d615e4f82)

The fix will be available with the upcoming 2.2.10 release.
